### PR TITLE
Introduced AbstractFeature

### PR DESCRIPTION
**Pull request by andreban**
- AbstractFeature allows removing boilerplate when implementing a Feature and only adding code for the fields we actually want to add details to.
    
- Updated AppsFlyerFeature and FirsRunFlagFeature to use the AbstractFeature.

### CODE REVIEW

File: packages/core/src/lib/features/AbstractFeature.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import {Feature} from './Feature';
+
+export class AbstractFeature implements Feature {
```
Comment (PEConn) [inline packages/core/src/lib/features/AbstractFeature.ts:19]: I'm not quite sure I'd call this AbstractFeature, to me "abstract" kinda means "somewhat implemented, but left some gaps for subclasses to implement". Following examples from the Chrome codebase, I'd probably call this "EmptyFeature". It reads quite nicely too, "AppsFlyerFeature extends EmptyFeature but with these bits filled in".

**Review by PEConn — COMMENTED** [review]
Code looks good, just a comment about the naming (also you have a typo in your first comment).

File: packages/core/src/lib/features/AbstractFeature.ts (hunk 1 above)
Comment (andreban) [reply packages/core/src/lib/features/AbstractFeature.ts:19]: SGTM - naming is hard :P

Comment (andreban) [reply packages/core/src/lib/features/AbstractFeature.ts]: Renamed.

**Review by PEConn — APPROVED** [review]

**Pull request merged**